src/test/java/org/apache/drill_web_test_framework/rest_api/data/RestBaseSteps.java

```diff
@@ -29,4 +42,78 @@ public static void setupREST() {
     RestAssured.baseURI = PropertiesConst.DRILL_HOST;
     RestAssured.port = PropertiesConst.DRILL_PORT;
   }
+
+  public static String getStringFromResource(String resource) {
+    try (InputStream file = RestBaseSteps.class.getClassLoader()
+        .getResourceAsStream(resource)) {
+      return IOUtils.toString(file, Charset.defaultCharset());
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static Boolean waitForCondition(Function<Boolean, Boolean> condition) {
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    while (stopwatch.elapsed(TimeUnit.SECONDS) < PropertiesConst.DEFAULT_TIMEOUT) {
+      if (condition.apply(null)) return true;
+    }
+    return false;
+  }
+
+  public static boolean isQueryRunning(String queryProfileAttribute, String attributeValue, SessionFilter sessionFilter) {
+    Response response = given()
+        .filter(sessionFilter)
+        .get("/profiles.json")
+        .then()
+        .extract()
+        .response();
+    for (HashMap profile : response.jsonPath().getList("runningQueries", HashMap.class)) {
```

Comment:
Streams look better here. Also, there is a String method "equalsIgnoreCase()"
```suggestion
    return response.jsonPath().getList("runningQueries", HashMap.class).stream()
        .filter(profile -> attributeValue.startsWith((String) profile.get(queryProfileAttribute)))
        .anyMatch(profile -> ((String) profile.get("state")).equalsIgnoreCase("running"));
```